src/Microsoft.VisualStudio.Services.Agent/HostContext.cs

```diff
@@ -26,53 +26,41 @@ public sealed class HostContext : IHostContext, IDisposable
     {
         private readonly ConcurrentDictionary<Type, object> _serviceInstances = new ConcurrentDictionary<Type, object>();
         private readonly ConcurrentDictionary<Type, Type> _serviceTypes = new ConcurrentDictionary<Type, Type>();
-        private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
         private TraceSource _trace;
         private ITraceManager _traceManager;
-        private string _hostType;
-        
-        public HostContext(string hostType)
+
+        public HostContext(string hostType, CancellationToken cancellationToken)
+            : this(
+                hostType: hostType,
+                logFile: Path.Combine(IOUtil.GetDiagPath(), StringUtil.Format("{0}_{1:yyyyMMdd-HHmmss}-utc.log", hostType, DateTime.UtcNow)),
+                cancellationToken: cancellationToken)
         {
-            if (string.IsNullOrEmpty(hostType))
-            {
-                throw new ArgumentNullException(nameof(hostType));
-            }
-            
-            _hostType = hostType;
+        }
+
+        public HostContext(string hostType, string logFile, CancellationToken cancellationToken)
```

Comment:
I added this overload so I could fix the log file path produced by the HostContext L0 tests to be consistent with every other test class. The production ctor calls this overload.